server/src/test/java/org/elasticsearch/cluster/service/ClusterApplierServiceTests.java

```diff
@@ -351,9 +351,14 @@ public void testClusterStateApplierCantSampleClusterState() throws InterruptedEx
                 clusterApplierService.state();
                 error.set(new AssertionError("successfully sampled state"));
             } catch (AssertionError e) {
-                if (e.getMessage().contains("should not be called by a cluster state applier") == false) {
-                    error.set(e);
+                final var message = e.getMessage();
+                if (message.contains("On the cluster applier thread")
+                    && message.contains("you must use ClusterChangedEvent#state()")
+                    && message.contains("almost certainly a bug to read the latest-applied state")
+                    && message.contains("the new state has been committed at this point but is not yet applied")) {
```

Comment:
It is surprising to see only parts of the message, is there a reason to check it this way?

Reply:
I feel bad copy-pasting a literal string from the production code to the test. Normally I do this to highlight the important parts in the test, although in this case most of the string is important.

Reply:
Is it worth introducing a constant with error message?

Reply:
That has the same problem, it makes it easy to change the message content in a way that loses some information we currently consider to be important.

Reply:
Ok, I've done that in 94c3b282506.